src/libsystemd/sd-bus/bus-error.c

```diff
@@ -355,6 +356,14 @@ _public_ int sd_bus_error_has_name(const sd_bus_error *e, const char *name) {
         return streq_ptr(e->name, name);
 }
 
+_public_ int _sd_bus_error_has_names(const sd_bus_error *e, const char *name, ...) {
+        if (!e || !e->name)
+                return 0;
+
+        char **names = strv_from_stdarg_alloca(name);
```

Comment:
uh, this is weird: both lvalue and rvalue allocated from the stack. supposedly that's not safe

Reply:
i think it would be nicer to just write the loop here ourselves, instead of using strv_contains()). i.e. no need to build an strv, if you can just iterate directly:

```c
va_start(name, ap);
for (;;) {
        const char *p;

        p = va_arg(ap, const char *p);
        if (!p) {
                va_end(ap);
                return false;
        }
        if (sd_bus_error_has_name(e, p)) {
                va_end(ap);
                return true;
        }
}
```

Reply:
Yeah, that is indeed better. I used a while loop instead to make this a bit shorter.